tokio-console/src/state/trace.rs

```diff
@@ -0,0 +1,79 @@
+use console_api as proto;
+use tracing_causality::{Span, Update};
+
+fn span_from_pb(proto: proto::consequences::Span) -> Span<u64> {
+    Span {
+        id: proto.span_id.unwrap().into(),
+        metadata: proto.metadata_id.unwrap().id,
+    }
+}
+
+pub(crate) fn updates_from_pb(
+    proto: proto::consequences::causality::Update,
+) -> Box<dyn Iterator<Item = Update<u64>>> {
+    use proto::consequences::causality::Update as ProtoUpdate;
+    use std::iter::{once, repeat};
+    match proto {
+        ProtoUpdate::Extant(proto::consequences::Extant {
+            cause: Some(cause),
+            direct_consequences,
+            indirect_consequences,
+        }) => {
+            let cause = span_from_pb(cause);
+            let direct_consequences = direct_consequences
+                .into_iter()
+                .zip(repeat(cause.clone()))
+                .map(
+                    move |(consequence, cause)| tracing_causality::Update::OpenDirect {
+                        cause: cause,
+                        consequence: span_from_pb(consequence),
+                    },
+                );
+            let indirect_consequences = indirect_consequences.into_iter().zip(repeat(cause)).map(
+                move |(consequence, cause)| tracing_causality::Update::NewIndirect {
```

Comment:
what is moved into these closures? AFAICT the `move` keyword is unneeded:
```suggestion
                    |(consequence, cause)| tracing_causality::Update::OpenDirect {
                        cause: cause,
                        consequence: span_from_pb(consequence),
                    },
                );
            let indirect_consequences = indirect_consequences.into_iter().zip(repeat(cause)).map(
                |(consequence, cause)| tracing_causality::Update::NewIndirect {
```
i'm assuming this was because `cause` was captured into the closures at one point before the `zip(repeat(cause))` thing?